extensions/schema-compare/src/test/schemaCompare.test.ts

```diff
@@ -111,3 +108,94 @@ describe('SchemaCompareResult.start', function (): void {
 		should.equal(result.targetEndpointInfo, undefined);
 	});
 });
+
+describe('SchemaCompareMainWindow.execute', function (): void {
+	before(() => {
+		mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
+		mockExtensionContext.setup(x => x.extensionPath).returns(() => '');
+		testContext = createContext();
+	});
+
+	beforeEach(async function (): Promise<void> {
+		testContext.apiWrapper.reset();
+	});
+
+	it('Should fail for failing Schema Compare service', async function (): Promise<void> {
+		let sc = new SchemaCompareTestService(testStateScmp.FAILURE);
+
+		testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
+		let result = new SchemaCompareMainWindow(testContext.apiWrapper.object, sc, mockExtensionContext.object);
+
+		await result.start(null);
+		let promise = new Promise(resolve => setTimeout(resolve, 5000)); // to ensure comparison result view is initialized
```

Comment:
If you're having to do this, then you should fix the code in SchemaCompareMainWindow for the result.start() to ONLY resolve when everything is done.

Reply:
+1; explicit waits in unit test code is an orange flag, especially since they're supposed to be fast and modular.  Doing it in each of these tests adds 20 guaranteed seconds to execution time, even if the result view initializes faster.

Reply:
That does make sense. I have updated the code accordingly and the tests are running way faster  :-) .

Reply:
thanks for getting this figured out! :)